### DISCUSSION TITLE
Guidance on Using the Updated ERC20Mock in OpenZeppelin for Stablecoin Project

### DISCUSSION THREAD

**Discussion by viniciuscoassis** [Cyfrin/foundry-full-course-cu#1579] [General]
Hello everyone,

I've encountered a challenge with the recent update to the ERC20Mock contract in OpenZeppelin. I'm currently working on a stablecoin project and am unsure how to properly utilize the new version of ERC20Mock. The documentation seems a bit unclear to me.

I'm curious about the approach others in the community have taken. Have you adopted the new version, or have you opted to use an older branch, such as the one Patrick uses in his tutorials? If the new version is the better choice, could someone point me to a guide or provide some tips on how to use it effectively? I'm feeling a bit stuck at the moment.

Any assistance or insights would be greatly appreciated. Thank you in advance for your help.

Best regards,
Vinicius Assis

**Comment by viniciuscoassis** [comment]
![Screenshot 2024-04-06 at 22 15 46](https://github.com/Cyfrin/foundry-full-course-f23/assets/106977892/b6824d28-3975-4478-aef3-cb083e0f54b1)
Just saw he advised to install by 
```forge install openzeppelin/openzeppelin-contracts@v4.8.3 --no-commit``` 
by anyways, Is there a good way to learn about openzeppelin recent updates?

**Reply by EngrPips** [reply]
Hey @viniciuscoassis , honestly i haven't seen a resources that teaches and guide on how to use Openzeppelin contracts upgrade myself, what i basically do is open up the previous version and the new version try compare them see what is difference and what the new change is and that is how i get to be able to use the upgraded contract much more effectively.

**Reply by tamermint** [reply]
hi @EngrPips, latest version is v5.0.2 - there is no need to install 4.8.3 - correct?

**Reply by EngrPips** [reply]
Hello @tamermint, You can use the latest version and check `https://github.com/Cyfrin/foundry-full-course-cu/blob/main/chronological-updates.md` whenever something that is expected to work doesn't work. The chronological updates give updates on what might need some little changes in the course codebase based on anything that has changed since the course was released
